common/v2/features/Ens/EnsDashboard.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useContext, useState } from 'react';
+import styled from 'styled-components';
+import { Heading } from '@mycrypto/ui';
+
+import { StoreContext } from 'v2/services';
+import { Spinner, DashboardPanel } from 'v2/components';
+import { DEFAULT_NETWORK } from 'v2/config';
+import { translateRaw } from 'v2/translations';
+import { BREAK_POINTS, SPACING } from 'v2/theme';
+import { useEffectOnce } from 'v2/vendor/react-use';
+import { isLabelHashENS } from 'v2/services/EthService/validators';
+import { EnsSubgraphService } from 'v2/services/ApiService/TheGraph';
+
+import { DomainTableEntry } from './types';
+import MyDomains from './MyDomains';
+import NoDomains from './NoEnsDomains';
+import EnsLogo from './EnsLogo';
+
+const DashboardWrapper = styled.div`
+  width: 100%;
+`;
+
+const DashboardSubHeader = styled(Heading)`
+  margin-top: ${SPACING.NONE};
+  font-weight: bold;
+  margin-bottom: ${SPACING.BASE};
+`;
+
+const StyledLayout = styled.div`
+  width: 960px;
+  @media (max-width: ${BREAK_POINTS.SCREEN_SM}) {
+    width: 100%;
+  }
+  .Layout-content {
+    padding: ${SPACING.NONE};
+  }
+`;
+
+const SpinnerContainer = styled.div`
+  height: 400px;
+  display: float;
+  align-items: center;
+  justify-content: center;
+`;
+
+interface MyDomainsDate {
+  data: DomainTableEntry[];
+  isFetched: boolean;
+}
+
+const defaultData: MyDomainsDate = {
+  data: [] as DomainTableEntry[],
+  isFetched: false
+};
+
+export default function EnsDashboard() {
+  const { accounts } = useContext(StoreContext);
+  const [fetchedEnsData, setEnsData] = useState(defaultData);
+
+  // Only use the accounts on the Ethereum mainnet network
+  const accountsEthereumNetwork = accounts.filter((acc) => acc.networkId === DEFAULT_NETWORK);
+
+  useEffectOnce(() => {
+    EnsSubgraphService.instance
+      .fetchOwnershipRecords(accountsEthereumNetwork)
+      .then((data: DomainTableEntry[]) => setEnsData({ data: data || [], isFetched: true }));
+  });
+
+  const domainEntries = fetchedEnsData.data.map((ownedENSEntry) => ({
+    ...ownedENSEntry,
+    domainName: ownedENSEntry.domainName,
+    readableDomainName: !isLabelHashENS(ownedENSEntry.domainName)
+      ? ownedENSEntry.domainName
+      : translateRaw('ENS_DOMAIN_UNKNOWN_NAME')
+  }));
```

Comment:
1. Please stay consistent with the naming. It seems that were using 'record', 'registration' and 'entry' for the ~same data.
2. This data processing will be rerun on every render (which we don't control). Do we really need both `ensData` and `domainEntries`?
- `domainName` already exists on the object
- we can add `readableDomainName` to the service response `readableDomainName: !isLabelHashENS(domainName) && domainName` and let `<MyDomains />` handle the default value.
3. if you still need to process the service response, best to create a tested pure function that you call before updating the state.